graphql/resolvers.go

```diff
@@ -3211,6 +3211,14 @@ func (r *taskResolver) VersionMetadata(ctx context.Context, obj *restModel.APITa
 	return apiVersion, nil
 }
 
+func (r *taskResolver) Patch(ctx context.Context, obj *restModel.APITask) (*restModel.APIPatch, error) {
+	apiPatch, err := r.sc.FindPatchById(*obj.Version)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Couldn't find a patch with id: `%s` %s", *obj.Version, err.Error()))
```

Comment:
We should do a check similar to https://github.com/evergreen-ci/evergreen/blob/41b7592f43d24458bb51170afecc2c5218a1a5f8/graphql/resolvers.go#L3600-L3610
to determine if the task has a patch before performing this query. 

Since some tasks might not have a patch we should avoid throwing an error since that is a valid scenario.

Reply:
Why do we still need the whole patch if we can disable the Schedule and Restart buttons on the /task/:id pages based on the requester?

Reply:
Just a matter of future proofing things. :) Its not necessary for this feature. But it might be useful in the future.

Reply:
sounds good. Let's make sure to not, in the future, accidentally call it on all tasks because then it can end up being expensive :)